packages/web-components/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,198 @@
+ * @public
+ */
+export class Tooltip extends FASTElement {
+  /**
+   * The attached element internals
+   */
+  public elementInternals = this.attachInternals();
+
+  /**
+   * The item ID
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: id
+   */
+  @attr
+  public id: string = uniqueId('tooltip-');
+
+  /**
+   * Set the delay for the tooltip
+   */
+  @attr({ converter: nullableNumberConverter })
+  public delay?: number;
+
+  /**
+   * The default delay for the tooltip
+   * @internal
+   */
+  private defaultDelay: number = 250;
+
+  /**
+   * Set the positioning of the tooltip
+   */
+  @attr
+  public positioning?: string;
+
+  /**
+   * The id of the anchor element for the tooltip
+   */
+  @attr
+  public anchor: string = '';
+
+  /**
+   * Reference to the anchor element
+   * @internal
+   */
+  private get anchorElement(): HTMLElement | null {
+    const rootNode = this.getRootNode();
+    return (rootNode instanceof ShadowRoot ? rootNode : document).getElementById(this.anchor ?? '');
+  }
+
+  /**
+   * Reference to the anchor positioning style element
+   * @internal
+   */
+  protected anchorPositioningStyleElement: HTMLStyleElement | null = null;
+
+  public constructor() {
+    super();
+    this.elementInternals.role = 'tooltip';
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    if (this.anchorElement) {
+      // @ts-expect-error - Baseline 2024
+      const anchorName = this.anchorElement.style.anchorName || `--${this.anchor}`;
+      this.anchorElement.setAttribute('aria-describedby', this.id);
+
+      if (SUPPORTS_ANCHOR_POSITIONING) {
+        // @ts-expect-error - Baseline 2024
+        this.anchorElement.style.anchorName = anchorName;
+        // @ts-expect-error - Baseline 2024
+        this.style.positionAnchor = anchorName;
+      } else {
+        // Provide style fallback for browsers that do not support anchor positioning
+        if (!this.anchorPositioningStyleElement) {
+          this.anchorPositioningStyleElement = document.createElement('style');
+          document.head.append(this.anchorPositioningStyleElement);
+        }
+
+        // Given a position with <direction>-<alignment> format, return the proper CSS properties
+        const [direction, alignment] = this.positioning?.split('-') ?? [];
+        let centerAlignment;
+
+        if (alignment === undefined) {
+          if (direction === 'above' || direction === 'below') {
+            centerAlignment = 'centerX';
+          }
+          if (direction === 'before' || direction === 'after') {
+            centerAlignment = 'centerY';
+          }
+        }
+
+        const directionCSS = {
+          above: `bottom: anchor(${anchorName} top);`,
+          below: `top: anchor(${anchorName} bottom);`,
+          before: `right: anchor(${anchorName} left);`,
+          after: `left: anchor(${anchorName} right);`,
+        }[direction];
```

Comment:
Like this expression! Should we also assign a default value to `directionCSS` (same for `alignementCSS`) just in case the `positioning` attribute has an invalid value.